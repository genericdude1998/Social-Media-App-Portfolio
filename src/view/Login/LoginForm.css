*{
  box-sizing: border-box;
  padding: 0px; 
}

.baseForm{
    height: 50vh;
    width: 100vw;
    min-width: 600px;
    align-items:center;
    display: flex;
    flex-direction: column;
    justify-content: center;
}
.baseLabel{
    color: black;
    margin: 10px;
    text-align: center;
    font-weight: bold;
}
.baseInput{
   margin:5px;
   border: 0px;
   border: 1px solid black;

}
.baseButton{
    margin:20px;
    height: 20px;
    width: 50px;
    border: 1px solid black;
    background-color: white;
}