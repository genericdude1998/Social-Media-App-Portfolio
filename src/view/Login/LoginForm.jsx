import React from 'react';
import styles from './LoginForm.css';
import PropTypes from 'prop-types';

const LoginForm = ({
    username,
    password,
    onChangeUsername,
    onChangePassword,
    onSubmitLogin,
}) => {
    return (
        <form className={styles.baseForm} onSubmit={onSubmitLogin(username, password)}>
            <label className={styles.baseLabel}>Username:</label>
            <input type='text' className={styles.baseInput} onChange={onChangeUsername}/>
            <label className={styles.baseLabel}>Password:</label>
            <input type='password'className={styles.baseInput} onChange={onChangePassword}/>
            <button type='submit' className={styles.baseButton}>Login</button>
        </form>
    );
}

LoginForm.propTypes = {
    onChangeUsername: PropTypes.func.isRequired,
    onChangePassword: PropTypes.func.isRequired,
}

export default LoginForm;
