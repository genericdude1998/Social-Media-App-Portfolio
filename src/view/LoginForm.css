*{
  box-sizing: border-box;
  padding: 0px; 
}

.baseForm{
    background-image: url('/assets/background-login.jpg');
    background-size: cover;
    height: 100vh;
    width: 100vw;
    align-items:center;
    display: flex;
    flex-direction: column;
    height: 100vh;
    justify-content: center;
}
.baseLabel{
    color: black;
    margin: 10px;
    text-align: center;
    font-weight: bold;
}
.baseInput{
   margin:5px;
   border: 0px;
   border: 1px solid black;

}
.baseButton{
    margin:20px;
    height: 20px;
    width: 50px;
    border: 1px solid black;
    background-color: white;
}

.light{

}

.dark{

}