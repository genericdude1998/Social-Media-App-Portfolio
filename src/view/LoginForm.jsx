import React from 'react';
import styles from './LoginForm.css'

const LoginForm = () => {
    const [username, setUsername] = React.useState('');
    const [password, setPassword] = React.useState('');

    return (
        <form className={styles.baseForm}>
            <label className={styles.baseLabel}>Username:</label>
            <input type='text' className={styles.baseInput}/>
            <label className={styles.baseLabel}>Password:</label>
            <input type='password'className={styles.baseInput}/>
            <button type='submit' className={styles.baseButton}>Login</button>
        </form>
    );
}

export default LoginForm;
