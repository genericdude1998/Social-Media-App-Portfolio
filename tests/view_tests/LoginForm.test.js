import React from 'react';
import { shallow } from 'enzyme';
import LoginForm from '../../src/view/Login/LoginForm';
import {mockOnChangeUsername, mockOnChangePassword} from '../mockValues';

let loginFormWrapper = shallow(
    <LoginForm
        onChangePassword= {mockOnChangePassword}
        onChangeUsername= {mockOnChangeUsername}
    />
);
it('should match the snapshot', () => {
    expect(loginFormWrapper).toMatchSnapshot();
});