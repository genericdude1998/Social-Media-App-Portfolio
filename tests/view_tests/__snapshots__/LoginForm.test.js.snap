// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<form
  className="baseForm"
>
  <label
    className="baseLabel"
  >
    Username:
  </label>
  <input
    className="baseInput"
    onChange={[MockFunction]}
    type="text"
  />
  <label
    className="baseLabel"
  >
    Password:
  </label>
  <input
    className="baseInput"
    onChange={[MockFunction]}
    type="password"
  />
  <button
    className="baseButton"
    type="submit"
  >
    Login
  </button>
</form>
`;
